pkg/plugins/plugins.go

```diff
@@ -199,17 +200,36 @@ func GetAll(imp string) ([]*Plugin, error) {
 		err error
 	}
 
-	chPl := make(chan plLoad, len(pluginNames))
+	driverExist := func(dn string) bool {
+		_, ok := storage.plugins[dn]
+		return ok
+	}
+
+	chPl := make(chan *plLoad, len(pluginNames))
+	var wg sync.WaitGroup
 	for _, name := range pluginNames {
+		if driverExist(name) {
+			chPl <- nil
```

Comment:
Here's the problem: at this level of the code, I don't think we can assume that the caller has the plugin that we are skipping here, and the caller should expect to get all the drivers for the implementation it specified, not just the ones that weren't already found before.
Then the caller can handle duplicates that it may already have cached.